### PR TITLE
Fixed an issue with native code throwing an exception expecting bound…

### PR DESCRIPTION
**Pull request by artakka**
…s to be a float array

Fixed same issue for center

Now both work:
       val bounds = floatArrayOf(-180.0f, -85.05113f, 180.0f, 85.05113f)
        tileSet.setBounds(*bounds) // or like tileSet.setBounds(-180.0f, -85.05113f, 180.0f, 85.05113f) doesn't matter
        val otherBounds : Array<Float> = bounds.toTypedArray()
        tileSet.setBounds(otherBounds)

### CODE REVIEW

**Comment by louwers** [comment]
@ovivoda I added a test and updated the changelog. Could you have a look and review this PR?

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt [hunk 1]
```diff
@@ -200,12 +205,14 @@ class TileSet(val tilejson: String, vararg tiles: String) {
...
      * @param center the Float array to set
      */
+    @Suppress("unused")
     fun setCenter(@Size(value = 2) vararg center: Float) {
-        this.center = center
+        val latLng = LatLng(center[1].toDouble(), center[0].toDouble())
+        setCenter(latLng)
     }
 
     fun setCenter(center: LatLng) {
```
Comment (ovivoda) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:214]: let's also add kotlin style comments/docs for this one as well

**Review by ovivoda — DISMISSED** [review]
It looks good. Nothing major to change. Tx @louwers !

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt [hunk 2]
```diff
@@ -200,12 +205,14 @@ class TileSet(val tilejson: String, vararg tiles: String) {
      *
      * @param center the Float array to set
```
[the remark this answers is not in the document]
Comment (louwers) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:206]: Why?

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt (hunk 1 above)
Comment (louwers) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:214]: There is no need to duplicate the documentation. The overloads will share the same documentation:

https://maplibre.org/maplibre-gl-native/android/api/-mapbox-g-l-android-s-d-k/com.mapbox.mapboxsdk.style.sources/-tile-set/index.html

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ovivoda) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:206]: just for the user of the method to know what to expect.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt (hunk 1 above)
Comment (ovivoda) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:214]: This is true, but for consistency I rather add the documentation here as well. I think, not adding it will encourage future PR's to not add documentation and it will be hard to follow every time if the method is overload or not.

Comment (louwers) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:214]: I disagree, since the behavior of both the overloads is the same. Duplicating the documentation will just make it harder to update and will only add noise. I think it will also show the documentation double, which is really not what we want in this case I think.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (louwers) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:206]: Isn't that an implementation detail? Since a double is wider it should not affect the user.

[the remark this answers is not in the document]
Comment (artakka) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:206]: Yes, it does not change anything for the users.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt (hunk 1 above)
Comment (artakka) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:214]: Will add a comment. I think we should deprecate setCenter that take two floats. It is not strictly typed and very easy to confuse which parameter is which. It does take longitude first, which is very confusing.  @ovivoda @louwers what do you think about marking as deprecated and steering them toward using LatLng or shall I leave it as is?

Comment (artakka) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:214]: I wish there was strongly typed alternative to setting the bounds too.

Comment (louwers) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:214]: > what do you think about marking as deprecated and steering them toward using LatLng 

@artakka Sounds good to me!

Comment (artakka) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/TileSet.kt:258]: @louwers @ovivoda Deprecated the unsafe setBounds and added a strongly typed alternative. Also added Doc comments to all of them.

Please check again.

**Review by louwers — APPROVED** [review]
